csi/util.go

```diff
@@ -118,11 +118,26 @@ func getVolumeOptions(volumeID string, volOptions map[string]string) (*longhornc
 
 		if isShared {
 			vol.AccessMode = string(longhorn.AccessModeReadWriteMany)
-		} else {
-			vol.AccessMode = string(longhorn.AccessModeReadWriteOnce)
 		}
 	}
 
+	if exclusive, ok := volOptions["exclusive"]; ok {
+		isExclusive, err := strconv.ParseBool(exclusive)
+		if err != nil {
+			return nil, errors.Wrap(err, "invalid parameter exclusive")
+		}
+		if isExclusive && vol.AccessMode == string(longhorn.AccessModeReadWriteMany) {
+			return nil, errors.New("cannot set both share and exclusive to true")
```

Comment:
Similar to migratable RWX volume (around [line 147](https://github.com/shikanime/longhorn-manager/blob/pr3236/csi/util.go#L147-L152)), we should just ignore the exclusive flag and log the issue here, rather than return an error to break the volume creation.

Reply:
I think we can introduce  a `storageclass` validation in https://github.com/longhorn/longhorn-manager/tree/master/webhook/resources as well as validate the two values in volume validating logic. Then, we don't need a complicated check here.